client/src/annoMatrix/annoMatrix.ts

```diff
@@ -629,49 +641,49 @@ Return cache keys for columns associated with this query.  May return
       //     ", "
       //   )}]`
       // );
+      // @ts-expect-error --- TODO revisit:
+      // `reduce`: This expression is not callable.
       this._cache[field] = toDrop.reduce(
```

Comment:
Hmm yeah interesting that TS is saying toDrop `Int32Array | (string | number)[]` is not callable 🤔 Feels like a TS bug 😂

Reply:
I need to spend some time looking through [this](https://github.com/microsoft/TypeScript/issues/44373) and [this](https://github.com/microsoft/TypeScript/issues/36390) but was thinking they might be applicable to a couple of suppressions I have added.

Reply:
Ooh thanks so much for research, Mim! Ahh I see, so it sounds like Int32Array and a regular array have difference reduce signatures, because Int32Array can guarantee `T` to be `number`, but regular array's `T` can be `number | string`, and the `array` arg type is different too `array: Int32Array` vs. `array: T[]`. So TS can't type anything for us in this case. But the error message `not callable` definitely seems misleading 😆 

```ts
// Int32Array
reduce(callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: Int32Array) => number, initialValue: number): number;

// Array
reduce(callbackfn: (previousValue: T, currentValue: T, currentIndex: number, array: T[]) => T, initialValue: T): T;
```